test/core/tests/tag_catch.py

```diff
@@ -121,7 +121,7 @@ def check_results(self, flow, checker):
                     data = task.data
                     got = sorted(m.value for m in task.metadata if m.type == "attempt")
                     if flow._graph[step.id].parallel_step:
-                        if "control" in task.id:
```

Comment:
TODO: Check task_metadata instead of checking of checking for the task-id string. the taskmetadata will contain the internal task type which can let us know if it's a control task or not.